File: backend/Business/Entities/Notification.cs
using Sotlaora.Business.Enums;
using Sotlaora.Business.Models;

namespace Sotlaora.Business.Entities
{
    public class Notification
    {
        public int Id { get; set; }
        public string Title { get; set; } = string.Empty;
        public string Message { get; set; } = string.Empty;
        public NotificationType Type { get; set; } // e.g., "urgent", "info", "setup_required"
        public bool IsRead { get; set; }
        public DateTime CreatedAt { get; set; }

        public int UserId { get; set; }
        public User User { get; set; } = null!;

        public NotificationMetadata Meta { get; set; } = new();
    }
}

File: frontend/types/Notification.ts
export interface NotificationDTO {
  title: string;
  message: string;
  type: NotificationType;
  slug: string;
}

export enum NotificationType {
    System = 0,
    Assigned = 1,
    NewOrder = 2,
    Completed = 3,
    Urgent = 4,
    SetupRequired = 5
}

export interface Notification {
  id: number;
  title: string;
  message: string;
  type: NotificationType;
  isRead: boolean;
  createdAt: string;
  slug: string;
  userId: number;
  meta: NotificationMetadata;
}

export interface NotificationMetadata {
  orderId?: number;
  clientName?: string;
  category?: string;
  amount?: string;
}